opencog/atomspace/AtomTable.cc

```diff
@@ -455,13 +459,6 @@ Handle AtomTable::add(AtomPtr atom, bool async)
     else if (atom == orig)
         atom = clone_factory(atom_type, atom);
 
-    // Lock before checking to see if this kind of atom is already in
-    // the atomspace.  Lock, to prevent two different threads from
-    // trying to add exactly the same atom.
-    std::unique_lock<std::recursive_mutex> lck(_mtx);
-    Handle hcheck(getHandle(orig));
-    if (hcheck) return hcheck;
-
     atom->copyValues(Handle(orig));
```

Comment:
This, the `copyValues`, can be safely moved out from under the lock.  Not sure about the rest.